File: GitToolBox/src/main/java/zielu/gittoolbox/ui/StatusText.java
package zielu.gittoolbox.ui;

import org.jetbrains.annotations.NotNull;
import zielu.gittoolbox.ResBundle;
import zielu.gittoolbox.status.GitAheadBehindCount;
import zielu.gittoolbox.status.Status;

public final class StatusText {
  private StatusText() {
    throw new IllegalStateException();
  }

  public static String format(@NotNull GitAheadBehindCount aheadBehind) {
    Status status = aheadBehind.status();
    if (status.isValid()) {
      if (status == Status.NO_REMOTE) {
        return ResBundle.getString("git.no.remote");
      } else {
        return StatusMessages.getInstance().aheadBehindStatus(aheadBehind);
      }
    } else {
      return ResBundle.getString("git.na");
    }
  }

  public static String formatToolTip(@NotNull GitAheadBehindCount aheadBehind) {
    if (aheadBehind.status() == Status.SUCCESS) {
      return "";
    } else {
      return StatusMessages.getInstance().aheadBehindStatus(aheadBehind);
    }
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/ui/StatusText.kt
package zielu.gittoolbox.ui

import zielu.gittoolbox.ResBundle
import zielu.gittoolbox.status.GitAheadBehindCount
import zielu.gittoolbox.status.Status

internal object StatusText {
  @JvmStatic
  fun format(aheadBehind: GitAheadBehindCount): String {
    val status = aheadBehind.status()
    return if (status.isValid()) {
      if (status === Status.NO_REMOTE) {
        ResBundle.message("git.no.remote")
      } else {
        StatusMessagesService.getInstance().aheadBehindStatus(aheadBehind)
      }
    } else {
      ResBundle.na()
    }
  }

  @JvmStatic
  fun format(extendedRepoInfo: ExtendedRepoInfo): String {
    val parts = ArrayList<String>()
    val changedCount = extendedRepoInfo.changedCount
    if (!changedCount.isEmpty()) {
      parts.add(ResBundle.message("change.count.x.changes.label", changedCount.value))
    }
    return parts.joinToString(" ")
  }

  @JvmStatic
  fun formatToolTip(aheadBehind: GitAheadBehindCount): String {
    return if (aheadBehind.status() === Status.SUCCESS) {
      ""
    } else {
      StatusMessagesService.getInstance().aheadBehindStatus(aheadBehind)
    }
  }
}
